From aef665ccf10ae1098fa804093a0d4250f009f694 Mon Sep 17 00:00:00 2001
From: Ofer Heifetz <oferh@marvell.com>
Date: Wed, 27 Jan 2016 10:51:02 +0200
Subject: [PATCH 0092/1345] serial: mvebu-uart: use shared irq

commit  d3c609dba3bafd6f56f74664b08f0e3821afc77e from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Driver used no flags for irq_request, this worked fine
for buildroot which opens single console but fails
for ubuntu which opens several consoles.

This commit enables IRQF_SHARED for this uart driver
irq_request.

Change-Id: I046c9bde6e17e8fdcc6dc3542054500e2c82d624
Signed-off-by: Ofer Heifetz <oferh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27690
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/mvebu-uart.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/tty/serial/mvebu-uart.c b/drivers/tty/serial/mvebu-uart.c
index 45b57c2..76969b6 100644
--- a/drivers/tty/serial/mvebu-uart.c
+++ b/drivers/tty/serial/mvebu-uart.c
@@ -574,7 +574,7 @@ static int mvebu_uart_probe(struct platform_device *pdev)
 	port->line       = 0; /* single port: force line number to  0 */
 
 	port->irq        = irq->start;
-	port->irqflags   = 0;
+	port->irqflags   = IRQF_SHARED;
 	port->mapbase    = reg->start;
 
 	port->membase = devm_ioremap_resource(&pdev->dev, reg);
-- 
1.7.9.5

